packages/ui-rating/src/Rating/index.tsx

```diff
@@ -77,38 +78,27 @@ class Rating extends Component<RatingProps> {
     const { valueNow, iconCount, valueMax } = this.props
 
     // prevent divide by zero errors
-    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
-    const max = valueMax > 0 ? valueMax : iconCount
+    const max = !!valueMax && valueMax > 0 ? valueMax : iconCount
```

Comment:
Adding the `!` at the end of iconCount will fix the tpye of `max` to be always be a number (since iconCount has a defaultValue of 3).

```suggestion
    const max = !!valueMax && valueMax > 0 ? valueMax : iconCount!
```